lib/board.rb

```diff
@@ -6,22 +6,13 @@ def initialize
   end
 
   def valid_placement?(ship, coordinates)
-    letter_by_itself = coordinates.map { |coordinate| coordinate[0] }
-    number_by_itself = coordinates.map { |coordinate| coordinate[1] }
-    letter_byself_same_value = if letter_by_itself.uniq.length == letter_by_itself.length
-    end
-    number_byself_same_value = if number_by_itself.uniq.length == number_by_itself.length
-    end
-    letter_order = ("A".."D").each_cons(ship.length) { |letter| p letter }
-    number_order = (1..4).each_cons(ship.length) { |number| p number }
-
     if coordinates.length == ship.length
-      # if the number of coordinates is equal to ship's length
-      if letter_byself_same_value && (number_by_itself == number_order.any?)
-      #if the letters are the same and the numbers are consecutive
-    elsif (letter_by_itself == letter_order.any?) && number_byself_same_value
-      # if the letters are consecutive and the numbers are the same
-      else false
+      letter_by_itself = coordinates.map { |coordinate| coordinate[0] }
+      number_by_itself = coordinates.map { |coordinate| coordinate[1].to_i }
+      if letter_by_itself.uniq.length == 1
+        number_by_itself.each_cons(2).all?{ |x, y| y == x + 1  }
+      elsif number_by_itself.uniq.length == 1
+        letter_by_itself.each_cons(2).all?{ |x, y| y.ord == x.ord + 1  }
```

Comment:
Ian did say to to read the whole iteration before beginning. Lesson learned. Would have been helpful to be tipped off on .each_cons and ordinal values. Learned the hard way on this one.